### PR TITLE
Account for macro invocation in `let mut $pat` diagnostic.

### LINKED ISSUE #65122: ICE: macro expansion using `expr` fragment as identifier
Regression from stable to nightly

```rust
macro_rules! mac {
    ($eval:expr) => {
        let mut $eval = ();
    };
}

fn foo() {
    mac! { does_not_exist!() }
}
```

backtrace
```
thread 'rustc' panicked at 'visit_mac disabled by default', src/libsyntax/mut_visit.rs:210:9
stack backtrace:
   0: backtrace::backtrace::libunwind::trace
             at /cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-0.3.37/src/backtrace/libunwind.rs:88
   1: backtrace::backtrace::trace_unsynchronized
             at /cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-0.3.37/src/backtrace/mod.rs:66
   2: std::sys_common::backtrace::_print_fmt
             at src/libstd/sys_common/backtrace.rs:76
   3: <std::sys_common::backtrace::_print::DisplayBacktrace as core::fmt::Display>::fmt
             at src/libstd/sys_common/backtrace.rs:60
   4: core::fmt::write
             at src/libcore/fmt/mod.rs:1028
   5: std::io::Write::write_fmt
             at src/libstd/io/mod.rs:1412
   6: std::sys_common::backtrace::_print
             at src/libstd/sys_common/backtrace.rs:64
   7: std::sys_common::backtrace::print
             at src/libstd/sys_common/backtrace.rs:49
   8: std::panicking::default_hook::{{closure}}
             at src/libstd/panicking.rs:196
   9: std::panicking::default_hook
             at src/libstd/panicking.rs:210
  10: rustc_driver::report_ice
  11: std::panicking::rust_panic_with_hook
             at src/libstd/panicking.rs:477
  12: std::panicking::begin_panic
  13: syntax::mut_visit::MutVisitor::visit_mac
  14: syntax::mut_visit::noop_visit_expr
  15: syntax::parse::parser::pat::<impl syntax::parse::parser::Parser>::parse_pat_with_range_pat
  16: syntax::parse::parser::pat::<impl syntax::parse::parser::Parser>::parse_pat_with_or
  17: syntax::parse::parser::pat::<impl syntax::parse::parser::Parser>::parse_top_pat
  18: syntax::parse::parser::stmt::<impl syntax::parse::parser::Parser>::parse_stmt_without_recovery
  19: syntax::parse::parser::stmt::<impl syntax::parse::parser::Parser>::parse_full_stmt
  20: syntax::ext::expand::<impl syntax::parse::parser::Parser>::parse_ast_fragment
  21: syntax::ext::mbe::macro_rules::ParserAnyMacro::make
  22: syntax::ext::expand::<impl syntax::ext::base::MacResult for syntax::ext::mbe::macro_rules::ParserAnyMacro>::make_stmts
  23: syntax::ext::expand::AstFragmentKind::make_from
  24: syntax::ext::expand::MacroExpander::fully_expand_fragment
  25: syntax::ext::expand::MacroExpander::expand_crate
  26: rustc_interface::passes::configure_and_expand_inner::{{closure}}
  27: rustc::util::common::time
  28: rustc_interface::passes::configure_and_expand_inner
  29: rustc_interface::passes::configure_and_expand::{{closure}}
  30: rustc_data_structures::box_region::PinnedGenerator<I,A,R>::new
  31: rustc_interface::queries::Query<T>::compute
  32: rustc_interface::queries::<impl rustc_interface::interface::Compiler>::expansion
  33: rustc_interface::interface::run_compiler_in_existing_thread_pool
  34: std::thread::local::LocalKey<T>::with
  35: syntax::with_globals
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.

error: internal compiler error: unexpected panic

note: the compiler unexpectedly panicked. this is a bug.

note: we would appreciate a bug report: https://github.com/rust-lang/rust/blob/master/CONTRIBUTING.md#bug-reports

note: rustc 1.40.0-nightly (2daa404e9 2019-10-02) running on x86_64-unknown-linux-gnu

note: compiler flags: -C codegen-units=1 -C debuginfo=2 --crate-type lib

note: some of the compiler flags provided by cargo are hidden
```

### PR DESCRIPTION
**Pull request by Centril**
Fixes https://github.com/rust-lang/rust/issues/65122.

r? @estebank

### CODE REVIEW

**Comment by estebank** [comment]
r=me when green

File: src/test/ui/parser/issue-65122-mac-invoc-in-mut-patterns.rs [hunk 1]
```diff
@@ -0,0 +1,26 @@
+// Regression test; used to ICE with 'visit_mac disabled by default' due to a
```
Comment (estebank) [inline src/test/ui/parser/issue-65122-mac-invoc-in-mut-patterns.rs]: `// check-only`?

Comment (Centril) [reply src/test/ui/parser/issue-65122-mac-invoc-in-mut-patterns.rs]: There is no such annotation to my knowledge.

**Comment by pnkfelix** [comment]
T-compiler discussed ; beta-accepted

**Pull request merged**